extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/TemplateDataBuildItem.java

```diff
@@ -55,35 +55,36 @@ final class TemplateDataBuildItem extends MultiBuildItem {
         this.properties = propertiesValue != null ? propertiesValue.asBoolean() : false;
     }
 
-    boolean isTargetAnnotatedType() {
-        return targetClass.asClass().name().equals(ValueResolverGenerator.TEMPLATE_DATA);
+    public boolean isTargetAnnotatedType() {
+        AnnotationValue targetValue = annotationInstance.value(ValueResolverGenerator.TARGET);
+        return targetValue == null || targetValue.asClass().name().equals(ValueResolverGenerator.TEMPLATE_DATA);
```

Comment:
I kinda expected `targetValue != null`, can you give me a quick hint how is type without that annotation annotation type, please?

Reply:
Yes, the `targetValue == null` means that the default value is used (`TemplateData.class` in this particular case). The default values are stored in the annotation defining class, not in the annotation instance. See also the `org.jboss.jandex.AnnotationInstance.valueWithDefault(IndexView, String)`  method and friends.

Reply:
And the `TemplateData.class` value means that a value resolver should be generated for the target of this annotation.

Reply:
Thanks